### ISSUE TITLE
Failed to add Jetstream stream with space in name

### ISSUE THREAD

**Issue by will-andrews-form3** [nats-io/nats.go#1218]
## Defect
New to using nats here so sorry if I'm getting confused.

For some reason I decided while playing around that I would use a stream name that contained a space and I was getting an error. After some play around and trying to get something working I started fresh but this time used a stream name without a space and it worked.

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [x] Included nats.go version
 - [x] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats.go` and the `nats-server` if one was involved:
Nats Server: 2.9.14
Nats.go: 1.23.0
#### OS/Container environment:
MacOS Docker
#### Steps or code to reproduce the issue:
```
func main() {
	nc, err := nats.Connect(serverURL)
	if err != nil {
		log.Fatalf("can't connect to nats: %s)", err)
	}

	defer nc.Close()

	js, err := nc.JetStream(nats.PublishAsyncMaxPending(256))
	if err != nil {
		log.Fatalf("can't create JS: %s", err)
	}

	_, err = js.AddStream(&nats.StreamConfig{
		Name:     "My stream",
		Subjects: []string{"something"},
	})

	if err != nil {
		log.Fatalf("failed to add stream: %s", err)
	}
}
```
#### Expected result:
A stream created

#### Actual result:
Error: `failed to add stream: context deadline exceeded`

Error in server: `[ERR] 172.17.0.1:35670 - cid:44 - "v1.23.0:go" - processPub Parse Error: "$JS.API.STREAM.CREATE.My stream _INBOX.dGbXX8mHjd6XYP7m8rhBeu.XtYjPX6R 238"`

**Comment by piotrpio** [comment]
Hello @will-andrews-form3! Thank you for submitting the issue - stream and consumer names cannot have spaces (https://docs.nats.io/nats-concepts/jetstream/streams#configuration), but we were missing this validation in the client and therefore you were getting a timeout instead of a proper error message. I have created a PR to fix that issue: #1219

**Comment by will-andrews-form3** [comment]
Amazing, thank you!
